sklearn/ensemble/gradient_boosting_cv.py

```diff
@@ -0,0 +1,877 @@
+                 presort='auto'):
+
+        self.n_stop_rounds = n_stop_rounds
+        self.score_precision = score_precision
+        self.max_iterations = max_iterations
+        self.cv = cv
+        self.n_iter_combo = n_iter_combo
+        self.scoring = scoring
+        self.refit = refit
+        self.n_jobs = n_jobs
+        self.pre_dispatch = pre_dispatch
+        self.verbose = verbose
+
+        self.loss = loss
+        self.learning_rate = learning_rate
+        self.subsample = subsample
+        self.min_samples_split = min_samples_split
+        self.min_samples_leaf = min_samples_leaf
+        self.min_weight_fraction_leaf = min_weight_fraction_leaf
+        self.max_depth = max_depth
+        self.init = init
+        self.random_state = random_state
+        self.max_features = max_features
+        self.max_leaf_nodes = max_leaf_nodes
+        self.presort = presort
+
+    def fit(self, X, y):
+        """Run fit with all sets of parameters till convergence.
+
+        X : array-like, shape = [n_samples, n_features]
+            Training vectors, where n_samples is the number of samples
+            and n_features is the number of features.
+        y : array-like, shape = [n_samples]
+            Target values (integers in classification, real numbers in
+            regression)
+            For classification, labels must correspond to classes.
+        """
+
+        params = self._get_params()
+        X, y = check_X_y(X, y, dtype=DTYPE)
+
+        cv = check_cv(self.cv, y)
+        if self.n_iter_combo is None:
+            param_iter = ParameterGrid(params)
+        else:
+            param_iter = ParameterSampler(params, self.n_iter_combo,
+                                          random_state=self.random_state)
+
+        parallel = Parallel(n_jobs=self.n_jobs, verbose=self.verbose,
+                            pre_dispatch=self.pre_dispatch)
+
+        out = parallel(delayed(_fit_single_param)
+                       (self._estimator_class, X, y, train, validation, params,
+                        self.n_stop_rounds, self.max_iterations, self.scoring,
+                        self.score_precision)
+                       for train, validation in cv.split(X)
+                       for params in param_iter)
+
+        n_splits = int(len(out)/len(param_iter))
+
+        grid_scores = []
+        i = 0
+        best_mean = -np.inf
+        for params in param_iter:
+
+            score_list = []
+            n_est_list = []
+            for idx in range(n_splits):
+                score, n_estimators = out[i]
+                n_est_list.append(n_estimators)
+                score_list.append(score)
+                i += 1
+
+            scores = np.array(score_list)
+            mean = np.mean(scores)
+            params['n_estimators'] = int(np.mean(n_est_list))
+            score_tuple = _CVScoreTuple(params, mean, scores)
+            grid_scores.append(_CVScoreTuple(params, mean, scores))
```

Comment:
Yes, I'd like to see a consistent interface like `GridSearchCV`'s results, and always had that in mind. I don't suppose we'll see it in 0.18. Sorry I wasn't fastidious enough to post this as a new issue months ago.